### PR TITLE
Added rotation to the labels

### PR DESCRIPTION
**Pull request by cvanelteren**
Hi,

Thanks for the nice package. I was missing the feature to manually control the angle of the label other than 0 degrees. This PR adds this feature as well as tidies up the code in `labelline.py` (i believe there is some unnecessary handling of properties going on).
Sincerely,
Casper

### CODE REVIEW

**Comment by cvanelteren** [comment]
Example: 
tmp
<details> <summary> code </summary>

```python
import proplot as plt, numpy as np


from labellines import labelLines

x = np.linspace(0, 1)
y = x
fig, ax = plt.subplots()
ax.plot(x, y, label="test")
labelLines(ax.get_lines(), align=False, rotation=90)
fig.show()
plt.show(block=1)
```

**Comment by cphyc** [comment]
Thanks for the contribution! Unfortunately, it looks like some tests are failing. I tried locally, and your update broke the logic. The issue is that the `get_rotation` function is called whenever the axis size changes (e.g. when `set_[xy]lim` is called). In your current implementation, it is only called once as far as I could tell.

You can test this locally with the following script:
```python
import matplotlib.pyplot as plt
import numpy as np

from labellines import labelLine

fig = plt.figure()
x = np.linspace(0, 10, 1000)
y = np.sin(x)

line, = plt.plot(x, y, label="$\sin(x)$")
labelLine(line, 2*np.pi, align=True)
plt.show()
```
[Screencast from 2023-09-27 13-58-00.webm](https://github.com/cphyc/matplotlib-label-lines/assets/5411875/69a9bd0d-4ff0-4e77-bbd9-df5fe00fe3f6)

**Comment by cvanelteren** [comment]
Cool test setup! Didn't know pytest could export to html like this.  The `LabelLine` object inherents from `Text` which has a default rotation implementation that should be compatible with this PR, what am I missing here? (see [here for the implementation]( https://github.com/matplotlib/matplotlib/blob/v3.8.0/lib/matplotlib/text.py#L284-L290) ).

**Comment by cphyc** [comment]
Maybe try having a look at `test_rotation_correction`. From the test artifacts (https://github.com/cphyc/matplotlib-label-lines/actions/runs/6325489737?pr=161), I see this test as failing more obviously.
image

**Comment by cphyc** [comment]
A bit more insight: your update breaks the piece of code that computes the rotation of the text to match the on-screen slope of the line. Since we're working in screen space, this depends on the size of the axis, which may have been updated since the LabelLines object got created.

So the `_rotation` property needs to be recomputed and cannot be cached (when we're using the `auto_align` keyword).

**Comment by cphyc** [comment]
They do indeed! Let me review the PR now then.

File: labellines/line_label.py [hunk 1]
```diff
@@ -96,6 +99,14 @@ def __init__(
 
         # Populate self._pos, self._anchor_a, self._anchor_b
         self._update_anchors()
+        if align and rotation is not None:
+            raise ValueError(
+                f"When rotation is set, align needs to be false or none was {align=}"
+            )
+        # if rotation is unset we assume 0
+        elif rotation == None:
```
Comment (cphyc) [inline labellines/line_label.py:107]:
```suggestion
        elif rotation is None:
```

File: labellines/line_label.py [hunk 2]
```diff
@@ -96,6 +99,14 @@ def __init__(
 
         # Populate self._pos, self._anchor_a, self._anchor_b
         self._update_anchors()
+        if align and rotation is not None:
+            raise ValueError(
+                f"When rotation is set, align needs to be false or none was {align=}"
```
Comment (cphyc) [inline labellines/line_label.py:104]:
```suggestion
                "When `rotation` is set, `align` needs to be set to False "
                f"or None, got {align=} instead."
```

File: labellines/line_label.py [hunk 3]
```diff
@@ -85,6 +86,8 @@ def __init__(
             If set to None, do not draw an outline, by default "auto".
         outline_width : float
             Width of the outline, by default 8.
+        rotation: Optional[float]
+            If set and align = False, controls the angle of the label
```
Comment (cphyc) [inline labellines/line_label.py:90]:
```suggestion
            If set and align = False, controls the angle of the label.
```

**Review by cphyc — COMMENTED** [review]
Thanks for the updates. You can update the linting for the moment, or if you want to, install it locally (`pip install pre-commit; pre-commit install`).

Would you mind also setting the default value of `align` to `None` so that the following code achieves the intended
```python
x = np.linspace(0, 10, 100)
y = np.sin(x)

line, = plt.plot(x, y, label="sin (x)")
labelLine(line, 6, rotation=45)
```
At the moment, this raises a ValueError exception.

**Comment by cvanelteren** [comment]
@cphyc I too hastily pushed the local commits. My bad! Current state should be good with changes to core as well. The default behavior is still aligning the labels with the curve (which is preferred anyway). I just ran into this issue recently when I wanted the labels turned 90 degrees ;-)!

**Comment by cvanelteren** [comment]
Thanks for cleaning up the mess @cphyc  ;-)!

**Pull request merged**